docs/UserGuide.md

```diff
@@ -11,117 +11,287 @@
+> The `<INDEX_TIMESLOT>` refers to the index number of the timeslot shown under each task. <br>
+> Both indexes **must be a positive integer** 1, 2, 3, ...
+
+Examples:
+* `confirm 3 2`
+
+#### Editing a task : `edit`
+Edits any component of a particular task <br>
+Format: `edit <INDEX> -n <TASK> -dt <START_DATE/TIME> to <END_DATE/TIME> -p <PRIORITY> -t <TAG(s)`
+
+> Edits the task at the specific `<INDEX>`. 
+> The index refers to the index number shown in the task list.
+> The index **must be a positive integer** 1, 2, 3, ... 
+>
+> Words in `UPPER_CASE` are the parameters. Other than `<INDEX>`, all parameters are optional. <br>
+> Order of parameters are **not** fixed.
+>
+> Time is in 24-hr format (e.g. 1330)
+
+Examples:
+* `edit 3 -n Meet John Tan -dt 08/10/2016 1000 to 1200 -p H -t friend`
+
+#### Editing a task by appending details to a task : `edit -ap`
+Edits a task by appending details to a particular task name <br>
+Format: `edit <INDEX> -ap <DETAILS_TO_APPEND_TO_TASK>`
+
+> Appends details to the task at the specific `<INDEX>`. 
+> The index refers to the index number shown in the task list.
+> The index **must be a positive integer** 1, 2, 3, ... 
+>
+> `<DETAILS_TO_APPEND_TO_TASK>` are words that the user wants to append to the back of the task.
+
+Examples:
+* `edit 3 -ap At UTown` edits `3. Meet John Doe` to `3. Meet John Doe At UTown`
+
+#### Editing a tag's name : `tag -e`
+Edits a tag’s name <br>
+Format: `tag -e <INDEX> <TAG>`
+
+> Edits the name of the tag at the specific `<INDEX>`. 
+> The index refers to the index number shown in the tag list.
+> The index **must be a positive integer** 1, 2, 3, ... 
+
+Examples:
+* `tag -e 5 Assignment`
+
+#### Deleting a tag : `tag -d`
+Deletes a particular tag <br>
+Format: `tag -d <INDEX>`
+
+> Deletes the tag at the specific `<INDEX>`
+> The index refers to the index number shown in the tag list.
+> The index **must be a positive integer** 1, 2, 3, ...
+
+Examples:
+* `tag -d 4` deletes the tag at Index 4
 
-#### Select a person : `select`
-Selects the person identified by the index number used in the last person listing.<br>
-Format: `select INDEX`
+#### Listing all tags : `tag -ls`
+Lists all tags in TARS <br>
+Format: `tag -ls`
 
-> Selects the person and loads the Google search page the person at the specified `INDEX`. 
-  The index refers to the index number shown in the most recent listing.<br>
-  The index **must be a positive integer** 1, 2, 3, ...
+#### Finding tags : `tag -f`
+Finds all tags by keywords (i.e. AND search) <br>
+Format: `tag -f <KEYWORD>[ , <KEYWORD>, <KEYWORD>, ...]`
+
+> `<KEYWORD>` are **case-insensitive**. The order of the `<KEYWORD>` does not matter.
+
+Examples:
+* `tag -f assignment` returns all tags with the word “assignment” (e.g. CS2103 assignment, CS2010 assignment)
+* `tag -f cs2103 assignment` returns all tags with both the words “cs2103” and “assignment" (e.g. CS2103 assignment)
+
+#### Marking tasks : `mark`
```

Comment:
you guys might want to consider to down the level of usability to as simple as <task id> done or done <task id> to have 2 levels might be abit too over for a simple marking of done and undone